pkg/packagekit/autoupdates.jsx

```diff
@@ -281,57 +281,51 @@ export default class AutoUpdates extends React.Component {
-                                           onChange={ t => this.handleChange(null, null, null, t) }>
-                                { hours.map(h => <Select.SelectEntry data={h + ":00"}>{('0' + h).slice(-2) + ":00"}</Select.SelectEntry>)}
-                            </Select.Select>
-                        </td>
-
-                        <td>{_("and restart the machine automatically.")}</td>
-                    </tr>
-                </table>
+                <div className="auto-conf">
+                    <span className="auto-conf-group">
+                        <Select.Select id="auto-update-type" enabled={!this.state.pending} initial={backend.type}
+                                       onChange={ t => this.handleChange(null, t, null, null) }>
+                            <Select.SelectEntry data="all">{_("Apply all updates")}</Select.SelectEntry>
+                            <Select.SelectEntry data="security">{_("Apply security updates")}</Select.SelectEntry>
+                        </Select.Select>
+                    </span>
+
+                    <span className="auto-conf-group">
+                        <Select.Select id="auto-update-day" enabled={!this.state.pending} initial={backend.day}
+                                       onChange={ d => this.handleChange(null, null, d, null) }>
+                            <Select.SelectEntry data="">{_("every day")}</Select.SelectEntry>
+                            <Select.SelectEntry data="mon">{_("on Mondays")}</Select.SelectEntry>
+                            <Select.SelectEntry data="tue">{_("on Tuesdays")}</Select.SelectEntry>
+                            <Select.SelectEntry data="wed">{_("on Wednesdays")}</Select.SelectEntry>
+                            <Select.SelectEntry data="thu">{_("on Thursdays")}</Select.SelectEntry>
+                            <Select.SelectEntry data="fri">{_("on Fridays")}</Select.SelectEntry>
+                            <Select.SelectEntry data="sat">{_("on Saturdays")}</Select.SelectEntry>
+                            <Select.SelectEntry data="sun">{_("on Sundays")}</Select.SelectEntry>
+                        </Select.Select>
+                    </span>
+
+                    <span className="auto-conf-group">
+                        <span className="auto-conf-text">{_("at")}</span>
+
+                        <Select.Select id="auto-update-time" enabled={!this.state.pending} initial={backend.time}
+                                       onChange={ t => this.handleChange(null, null, null, t) }>
+                            { hours.map(h => <Select.SelectEntry data={h + ":00"}>{('0' + h).slice(-2) + ":00"}</Select.SelectEntry>)}
+                        </Select.Select>
+                    </span>
+
+                    <span className="auto-conf-group auto-conf-text">{_("and restart the machine automatically.")}</span>
+                </div>
```

Comment:
We should not use tables for layout; they should be used only for tabular data.

Changing it to spans in divs lets us do special things with the layout.

Reply:
Ack, I'll remember for the future. I grew up with tables being the main (and often only) way of arranging things in a grid in HTML :)